vendor/k8s.io/kubernetes/pkg/volume/iscsi/iscsi_util.go

```diff
@@ -90,32 +90,39 @@ func updateISCSINode(b iscsiDiskMounter, tp string) error {
 type StatFunc func(string) (os.FileInfo, error)
 type GlobFunc func(string) ([]string, error)
 
-func waitForPathToExist(devicePath string, maxRetries int, deviceTransport string) bool {
+func waitForPathToExist(devicePath *string, maxRetries int, deviceTransport string) bool {
 	// This makes unit testing a lot easier
 	return waitForPathToExistInternal(devicePath, maxRetries, deviceTransport, os.Stat, filepath.Glob)
 }
 
-func waitForPathToExistInternal(devicePath string, maxRetries int, deviceTransport string, osStat StatFunc, filepathGlob GlobFunc) bool {
-	for i := 0; i < maxRetries; i++ {
-		var err error
-		if deviceTransport == "tcp" {
-			_, err = osStat(devicePath)
-		} else {
-			fpath, _ := filepathGlob(devicePath)
-			if fpath == nil {
-				err = os.ErrNotExist
+func waitForPathToExistInternal(devicePath *string, maxRetries int, deviceTransport string, osStat StatFunc, filepathGlob GlobFunc) bool {
+	if devicePath != nil {
+		for i := 0; i < maxRetries; i++ {
+			var err error
+			if deviceTransport == "tcp" {
+				_, err = osStat(*devicePath)
+			} else {
+				fpath, _ := filepathGlob(*devicePath)
+				if fpath == nil {
+					err = os.ErrNotExist
+				} else {
+					// There might be a case that fpath contains multiple device paths if
+					// multiple PCI devices connect to same iscsi target. We handle this
+					// case at subsequent logic. Pick up only first path here.
+					*devicePath = fpath[0]
+				}
 			}
+			if err == nil {
+				return true
+			}
+			if err != nil && !os.IsNotExist(err) {
```

Comment:
How did no one point out this `err != nil` is redundant? Don't have to fix it here, but we should fix this upstream.